nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-properties-loader/src/main/java/org/apache/nifi/properties/NiFiPropertiesLoader.java

```diff
@@ -241,9 +239,53 @@ public NiFiProperties load(String path) {
      */
     public NiFiProperties get() {
         if (instance == null) {
-            instance = loadDefault();
+            instance = setSensitivePropertiesKey(loadDefault());
```

Comment:
if `isKeyGenerationRequired` returns false, `setSensitivePropertiesKey` will have no effect. Maybe renaming the method to something like `setSensitivePropertiesKeyIfBlank` or externalizing the `if ` enclosing `setSensitivePropertiesKey`'s body would provide more readability ?

Reply:
Thanks for the suggestion, will refactor to externalize the conditional for better readability.